packages/core/src/shader/Shader.ts

```diff
@@ -46,30 +59,50 @@ export class Shader {
   static create(name: string, subShaders: SubShader[]): Shader;
 
   static create(
-    name: string,
-    vertexSourceOrShaderPassesOrSubShaders: SubShader[] | ShaderPass[] | string,
+    nameOrGalaceanShader: string,
+    vertexSourceOrShaderPassesOrSubShaders?: SubShader[] | ShaderPass[] | string,
     fragmentSource?: string
   ): Shader {
-    const shaderMap = Shader._shaderMap;
-    if (shaderMap[name]) {
-      throw `Shader named "${name}" already exists.`;
-    }
     let shader: Shader;
-    if (typeof vertexSourceOrShaderPassesOrSubShaders === "string") {
-      const shaderPass = new ShaderPass(vertexSourceOrShaderPassesOrSubShaders, fragmentSource);
-      shader = new Shader(name, [new SubShader("Default", [shaderPass])]);
+    const shaderMap = Shader._shaderMap;
+
+    if (!vertexSourceOrShaderPassesOrSubShaders) {
+      if (!this._shaderLab) {
+        throw "ShaderLab not been setted up";
+      }
+      // TODO: render state
+      const shaderInfo = this._shaderLab.parseShader(nameOrGalaceanShader);
+      const subShaderList = shaderInfo.subShaders.map((subshader) => {
+        const passList = subshader.passes.map((pass) => {
```

Comment:
`subshader`->`subShader`, i think you should install spell-checker

Reply:
fixed